backend/dataall/core/environment/services/managed_iam_policies.py

```diff
@@ -19,17 +22,54 @@ def __init__(
         self.environmentUri = environmentUri
         self.resource_prefix = resource_prefix
 
+    @abstractmethod
+    def generate_policy_name(self) -> str:
+        """
+        Returns string and needs to be implemented in the ManagedPolicies inherited classes
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def generate_empty_policy(self) -> dict:
+        """
+        Returns dict and needs to be implemented in the ManagedPolicies inherited classes
+        """
+        raise NotImplementedError
+
+
+class PolicyManager(object):
+    def __init__(
+            self,
+            role_name,
+            account,
+            environmentUri,
+            resource_prefix,
+    ):
+        self.role_name = role_name
+        self.account = account
+        self.environmentUri = environmentUri
+        self.resource_prefix = resource_prefix
+        self.ManagedPolicies = ManagedPolicy.__subclasses__()
```

Comment:
* As we discussed offline `__subclasses__` will return only those classes that have imported in the interpreter prior to this call. How do we ensure that this happens?
* Since self.ManagedPolicies isn't used anywhere else in the class I wouldn't make it a member but a local var. Also I'd most likely name in `ManagedPoliciesCls`

Reply:
- we can ensure that it is initialized by adding it to the `__init__` of `dataall/modules/dataset_sharing/__init__.py` using the `(ModuleInterface)`